cpp_faster_fifo/cpp_lib/faster_fifo.cpp

```diff
@@ -145,28 +145,42 @@ bool timer_positive(const struct timeval &timer) {
     return (timer.tv_sec > 0) || (timer.tv_sec == 0 && timer.tv_usec > 0);
 }
 
-int queue_put(void *queue_obj, void *buffer, void *msg_data, size_t msg_size, int block, float timeout) {
+int queue_put(void *queue_obj, void *buffer, const void **msgs_data, const size_t *msg_sizes, const size_t num_msgs, const int block, const float timeout) {
```

Comment:
Having `const <PID type>` in the function interface is something not everyone agrees with. You're copying the value one way or another, so it does not make any difference for the caller. What you end up doing is exposing the implementation detail of the function (constness of some local variables passed by value) through the function interface.

I'm fine with it though, don't care much one way or the other. I've met adepts of const-correctness who like doing this.

`const void *` and other buffers make sense though!